### PR TITLE
mesa: 19.2.1 -> 19.2.3 (and associated fixes)

### PR DESCRIPTION
**Pull request by adisbladis**
###### Motivation for this change
This was released the other day, and I want to use it on my AMD gpu.

A lot of things have changed upstream, `mesa` stopped providing a bunch of headers as `libglvnd` also provides the same headers (though not _exactly_ the same), so we need a bit of fixups for some packages to build.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that Affected packagesrelevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @vcunat 

###### New failures
None

###### Still failing (ie failing since before this PR)
- python38Packages.pysonos
- appeditor
- liquidfun
- ~python27Packages.neurotools~ - Package dropped, thanks @nico202 
A dependency is python3 only while neurotools is python2 only
cc maintainer @nico202 
- python37Packages.tensorflow
- python37Packages.edward
Pulls in tensorflow

###### Unknown
- qt59.qtwebkit
I don't want to waste time on this package, it's not really used by anything
A follow-up PR removing Qt 5.9 will be created

##### Notes
- libGLU_combined should be removed in a follow-up PR

### CODE REVIEW

File: pkgs/servers/x11/xorg/default.nix [hunk 1]
```diff
@@ -2691,13 +2691,23 @@ lib.makeScope newScope (self: with self; {
     meta.platforms = stdenv.lib.platforms.unix;
   }) {};
 
-  xorgserver = callPackage ({ stdenv, pkgconfig, fetchurl, xorgproto, openssl, libX11, libXau, libXaw, libxcb, xcbutil, xcbutilwm, xcbutilimage, xcbutilkeysyms, xcbutilrenderutil, libXdmcp, libXfixes, libxkbfile, libXmu, libXpm, libXrender, libXres, libXt }: stdenv.mkDerivation {
```
Comment (worldofpeace) [inline pkgs/servers/x11/xorg/default.nix]: This file is generated, so you need to move your changes into `overrides.nix` in the same directory.

**Review by worldofpeace — CHANGES_REQUESTED** [review]
Just one comment, obviously haven't tested anything yet.
The fixes to the GNOME/Gtk side of things are appreciated :sparkles:

**Comment by worldofpeace** [comment]
> Still failing:
> * appeditor

It actually fails on master
```
../src/DesktopApp.vala:133.28-133.32: error: Invalid assignment from owned expression to unowned variable
            unowned string _path = info.get_string (KeyFileDesktop.KEY_PATH);
                           ^^^^^
../src/DesktopApp.vala:134.17-134.21: error: The name `_path' does not exist in the context of `AppEditor.DesktopApp.get_path'
            if (_path == null) {
                ^^^^^
../src/DesktopApp.vala:135.17-135.21: error: The name `_path' does not exist in the context of `AppEditor.DesktopApp.get_path'
                _path = "";
                ^^^^^
../src/DesktopApp.vala:138.20-138.24: error: The name `_path' does not exist in the context of `AppEditor.DesktopApp.get_path'
            path = _path;
                   ^^^^^
```

I can fix this since I maintain it :+1:

**Comment by adisbladis** [comment]
> It actually fails on master 

I should have made it more clear :) What I mean by "still failing" is that it's failing since before the PR.

File: pkgs/servers/x11/xorg/default.nix (hunk 1 above)
Comment (adisbladis) [reply pkgs/servers/x11/xorg/default.nix]: Thanks for pointing that out :+1:

**Comment by worldofpeace** [comment]
> > It actually fails on master
> 
> I should have made it more clear :) What I mean by "still failing" is that it's failing since before the PR.

Ahh, I see :smile: . Fixed that one in a6d3df106c0db76d6186847a1d203b840ab01143.

**Comment by adisbladis** [comment]
Note: This PR is currently only tested on Linux. I'm not sure how this affects MacOS.

**Comment by nico202** [comment]
I'm sorry but I'm not using neurotools anymore, and it is not a dependency of anything. Since python2 is at EOL and neurotools has not been updated, might we drop it?

**Comment by adisbladis** [comment]
> I'm sorry but I'm not using neurotools anymore, and it is not a dependency of anything. Since python2 is at EOL and neurotools has not been updated, might we drop it?

Sounds good to me. Could you make a PR?

**Comment by nico202** [comment]
@adisbladis: done

**Comment by adisbladis** [comment]
If there is no objection I will merge this tomorrow (sunday).

I don't believe we will see extensive testing of this PR by anyone else and there are no new build failures, so I feel pretty confident.

cc mesa maintainer @vcunat

**Comment by adisbladis** [comment]
@GrahamcOfBorg build xorg.xorgserver

**Comment by jonringer** [comment]
Is this for navi cards? because I would love to use my 5700XT xD

**Comment by adisbladis** [comment]
@jonringer Hehe :) I'm using this from my 5700XT and it works great!

**Comment by adisbladis** [comment]
I have now tested this on MacOS too, I didn't find any new broken packages there either, but I have not run a full rebuild.

**Comment by jonringer** [comment]
wish i was at home with my 3900X to review this, on my laptop. If it's not merged by tomorrow, I'll review this :)

**Comment by adisbladis** [comment]
Pushed a patch to fix the `qtbase` build on Darwin and bumped the default version of Qt from 5.11 to 5.12 on Darwin.

This is required as Qt 5.11 no longer builds, and I'd rather spend my time fixing Qt 5.12 than going down the rabbit hole of patching old unsupported software.

**Comment by adisbladis** [comment]
After a full `nox-review` on `x86_64-linux` (with only the failures listed in this PR) and a partial but fairly extensive Darwin rebuild (most graphical packages don't build on Darwin to begin with) I feel pretty good about this PR.

**Comment by eadwu** [comment]
https://github.com/NixOS/nixpkgs/pull/73402

**Comment by FRidh** [comment]
19.2.4 is now also in staging-next

**Pull request merged**